schemas/test/en/test_supplementary_data.json

```diff
@@ -22,7 +22,8 @@
         },
         {
             "name": "sds_dataset_id",
-            "type": "string"
+            "type": "string",
+            "optional": true
```

Comment:
Can't recall what we said about this - it was for the test case right? Can it now be reverted? I think if the sds_dataset_id is exists for a given form type it won't be optional

Reply:
Yeah I added this for coverage of the case where an sds_dataset_id is provided, the questionnaire store state is saved, and then its loaded and the new metadata doesn't contain an id, and so the old data needs deleting - if thats a case that will never happen I can revert this and slightly tweak the code not to worry about that scenario 👍 thoughts?

Reply:
Yeah I can understand checking for when the prepop data for a given id has been changed, but I'm not sure about the scenario above, as the form type would still require that sds_dataset_id, but maybe worth leaving this for now and double checking this with @MebinAbraham

Reply:
A form type will either have it or not. There won't ever be a "maybe" case, so we don't need this?. If form type that had it, won't remove it midway. No reqs for that. Only for changing datasets.

Reply:
Removed the optional key and the code that was accounting for this and testing it 👍 
That makes sense